### PR TITLE
Fix pending owner late overwrite issue

### PR DESCRIPTION
**Pull request by ericnordelo**

### CODE REVIEW

File: packages/access/src/ownable/ownable.cairo [hunk 1]
```diff
@@ -273,6 +273,8 @@ pub mod OwnableComponent {
         fn _transfer_ownership(
             ref self: ComponentState<TContractState>, new_owner: ContractAddress
         ) {
+            self.Ownable_pending_owner.write(Zero::zero());
+
```
[the remark this answers is not in the document]
Comment (andrew-fleming) [reply packages/access/src/ownable/ownable.cairo:277]: With that, do we even need `_accept_ownersip` anymore with this removal? It's rereading the pending owner and calling `_transfer_ownership` when invoked from `accept_ownership`...AFAICT it makes more sense to change `_accept_ownership` to `_transfer_ownership` in the external `accept_ownership`

```diff
fn accept_ownership(ref self: ComponentState<TContractState>) {
    let caller = get_caller_address();
    let pending_owner = self.Ownable_pending_owner.read();
    assert(caller == pending_owner, Errors::NOT_PENDING_OWNER);
-   self._accept_ownership();
+   self._transfer_ownership(pending_owner);
}
```

File: packages/access/src/tests/test_ownable.cairo [hunk 2]
```diff
@@ -86,6 +86,20 @@ fn test__transfer_ownership() {
...
+    let mut state = setup();
+
+    state.Ownable_pending_owner.write(OTHER());
+    let current_pending_owner = state.Ownable_pending_owner.read();
+    assert_eq!(current_pending_owner, OTHER());
+
+    state._transfer_ownership(RECIPIENT());
+
+    let current_owner = state.Ownable_pending_owner.read();
+    assert!(current_owner.is_zero());
```
Comment (immrsd) [inline packages/access/src/tests/test_ownable.cairo:100]:
```suggestion
    let current_pending_owner = state.Ownable_pending_owner.read();
    assert!(current_pending_owner.is_zero());
```

File: packages/access/src/tests/test_ownable_twostep.cairo [hunk 3]
```diff
@@ -244,6 +244,22 @@ fn test_renounce_ownership() {
...
+    start_cheat_caller_address(contract_address, OWNER());
+
+    state.Ownable_pending_owner.write(OTHER());
+    let current_pending_owner = state.Ownable_pending_owner.read();
+    assert_eq!(current_pending_owner, OTHER());
+
+    state.renounce_ownership();
+
+    let current_owner = state.Ownable_pending_owner.read();
+    assert!(current_owner.is_zero());
```
Comment (immrsd) [inline packages/access/src/tests/test_ownable_twostep.cairo:260]:
```suggestion
    let current_pending_owner = state.Ownable_pending_owner.read();
    assert!(current_pending_owner.is_zero());
```

File: packages/access/src/tests/test_ownable_twostep.cairo [hunk 4]
```diff
@@ -307,31 +323,6 @@ fn test_full_two_step_transfer() {
...
-    assert!(state.owner().is_zero());
-
-    start_cheat_caller_address(contract_address, OTHER());
-    state.accept_ownership();
-
-    spy.assert_only_event_ownership_transferred(contract_address, ZERO(), OTHER());
-    assert_eq!(state.owner(), OTHER());
-    assert!(state.pending_owner().is_zero());
-}
-
```
Comment (immrsd) [inline packages/access/src/tests/test_ownable_twostep.cairo:334]: Oh, there was a test case ensuring the exact scenario is possible?

**Review by immrsd — APPROVED** [review]
Looks good

**Review by andrew-fleming — COMMENTED** [review]
Looking good! I left a couple comments

File: packages/access/src/tests/test_ownable_twostep.cairo (hunk 4 above)
Comment (ericnordelo) [reply packages/access/src/tests/test_ownable_twostep.cairo:334]: Yes it was 😞

File: packages/access/src/ownable/ownable.cairo (hunk 1 above)
[the remark this answers is not in the document]
Comment (ericnordelo) [reply packages/access/src/ownable/ownable.cairo:277]: Good call!

**Review by andrew-fleming — APPROVED** [review]
LGTM! I'd also include the removal of `_accept_ownership` to the changelog

**Pull request merged**